api/sso/sso_api/endpoints/password_change.py

```diff
@@ -1,36 +1,34 @@
-# Mycroft Server - Backend
-# Copyright (C) 2019 Mycroft AI Inc
-# SPDX-License-Identifier: 	AGPL-3.0-or-later
+#  Mycroft Server - Backend
+#  Copyright (c) 2022 Mycroft AI Inc
+#  SPDX-License-Identifier: 	AGPL-3.0-or-later
+#  #
+#  This file is part of the Mycroft Server.
+#  #
+#  The Mycroft Server is free software: you can redistribute it and/or
+#  modify it under the terms of the GNU Affero General Public License as
+#  published by the Free Software Foundation, either version 3 of the
+#  License, or (at your option) any later version.
+#  #
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU Affero General Public License for more details.
+#  #
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
-# This file is part of the Mycroft Server.
-#
-# The Mycroft Server is free software: you can redistribute it and/or
-# modify it under the terms of the GNU Affero General Public License as
-# published by the Free Software Foundation, either version 3 of the
-# License, or (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-# along with this program. If not, see <https://www.gnu.org/licenses/>.
+"""Defines the password change endpoint for the account API."""
 
-from binascii import a2b_base64
-from http import HTTPStatus
+from selene.api.endpoints import PasswordChangeEndpoint as CommonPasswordChangeEndpoint
 
-from selene.api import SeleneEndpoint
-from selene.data.account import AccountRepository
 
+class PasswordChangeEndpoint(CommonPasswordChangeEndpoint):
+    """Adds authentication to the common password changing endpoint."""
 
-class PasswordChangeEndpoint(SeleneEndpoint):
-    def put(self):
-        account_id = self.request.json["accountId"]
-        coded_password = self.request.json["password"]
-        binary_password = a2b_base64(coded_password)
-        password = binary_password.decode()
-        acct_repository = AccountRepository(self.db)
-        acct_repository.change_password(account_id, password)
+    @property
+    def account_id(self):
+        """Returns the account passed to the endpoint in the PUT request"""
+        return self.request.json["accountId"]
 
-        return "", HTTPStatus.NO_CONTENT
+    def _authenticate(self):
+        """Skips authentication because user is not logged in when this is called."""
```

Comment:
Why is the account not logged in when this is called?

Reply:
This is in the single sign on API.  can't authenticate if you aren't logged in yet 😄